src/utils/env.ts

```diff
@@ -65,3 +76,20 @@ export const lookupPcEnv = (pcEnvTld: string, lenient = false, envTlds: string[]
 
     return null;
 };
+
+/**
+ * Attempts to locate a PC environment corresponding to the provided url
+ * @param url A string representing the Genesys Cloud environment top-level domain to search for
+ * @returns A Genesys Cloud environment object if found; null otherwise.
+ */
+export const lookupGcEnv = (url: string, targetEnv: string, envList: Environment[] = GC_DEV_EXTRA_ENVS): PcEnv|null => {
+    const hostLocation = new URL(url);
```

Comment:
this could fail to parse.  For instance if url is not absolute.  Looks like we would technically be fine to throw the error all the way up, but, given we can already return null from this function and we present a user friendly message, maybe we should add a try/catch here and return null instead of throwing?